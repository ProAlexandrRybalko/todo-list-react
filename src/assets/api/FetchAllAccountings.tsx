import axios from "axios";
import { AccountingsSum } from "../../components/Accounting/AccountingsType";

export const getAllAccountings = () => axios.get<AccountingsSum>("http://localhost:8000/allAccountings");
