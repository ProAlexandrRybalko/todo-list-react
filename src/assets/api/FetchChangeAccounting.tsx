import axios from "axios";

export const changeAccounting = (id: string, inputWhere: string, inputHowMuch: number) => axios.patch<string>("http://localhost:8000/changeAccounting", 
    {
        id,
        data: { 
            where: inputWhere,
            howMuch: inputHowMuch 
        }
    }
);