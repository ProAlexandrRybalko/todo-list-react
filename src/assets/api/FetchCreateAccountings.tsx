import axios from "axios";
import { AccType, AccTypeId } from "../../components/Accounting/AccountingsType";

export const createAccounting = ({ where, date, howMuch }: AccType) => axios.post<AccTypeId>("http://localhost:8000/createAccountings", 
    {
        where, 
        date, 
        howMuch
    }
);

