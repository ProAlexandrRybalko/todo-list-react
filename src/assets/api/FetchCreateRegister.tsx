import axios from "axios";

export const setRegister = (fullName: string, email: string, password: string) => axios.post<string>("http://localhost:8000/register", 
    {
        fullName,
        email,
        password
    }
);

