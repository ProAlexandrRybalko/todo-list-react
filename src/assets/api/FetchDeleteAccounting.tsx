import axios from "axios";

export const deleteAccounting = (id: string) => axios.delete<string>("http://localhost:8000/deleteAccounting", 
    {
        data: { id }
    }
);
