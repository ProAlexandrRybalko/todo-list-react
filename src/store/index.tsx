import { TypedUseSelectorHook, useSelector } from "react-redux";
import { createStore, combineReducers, applyMiddleware } from "redux";
import { accountingsReducer } from "./AccountingsReducer";
import { sumReducer } from "./sumReducer";
import createSagaMiddleware from 'redux-saga';
import { rootWatcher } from "../saga";

const sagaMiddleware = createSagaMiddleware();

const rootReducer = combineReducers({
    sumReducer, 
    accountingsReducer
})

export const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(rootWatcher);

type RootState = ReturnType<typeof rootReducer>
export const useTypedSelector: TypedUseSelectorHook<RootState> = useSelector;
