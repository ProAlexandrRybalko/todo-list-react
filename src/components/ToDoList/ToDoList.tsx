import { useEffect } from "react";
import { MainWrapper } from "../../assets/styles/ToDoStyles"
import Header from "../../assets/styles/HeaderStyles";
import Content from "../../assets/styles/ContentStyles";
import Managing from "../../assets/styles/ManagingStyles";
import InputAccounting from "../../assets/styles/InputStyles";
import InputInfoAccounting from "../../assets/styles/InputInfoStyles";
import Sum from "../../assets/styles/SumStyles";
import ButtonAccounting from "../buttonAccounting/ButtonAccounting";
import ViewAccounting from "../../assets/styles/ViewAccountingStyles";
import Accounting from "../Accounting/Accounting";
import { useDispatch } from "react-redux";
import { useTypedSelector } from "../../store";
import { asyncSetAccountingsCreator } from "../../store/AccountingsReducer";
import { asyncSetSumCreator } from "../../store/sumReducer";

const ToDoList = () => {

    const dispatch = useDispatch();
    const accountings = useTypedSelector(state => state.accountingsReducer.accountings);
    const sum = useTypedSelector(state => state.sumReducer.sum);
    
    useEffect(() => {
        dispatch(asyncSetAccountingsCreator());
        dispatch(asyncSetSumCreator());
    }, []);

    return (
        <MainWrapper data-testid="todoList">
            <Header>
                <p>Учёт моих расходов</p>
            </Header>
            <Content>
                <Managing>
                    <InputInfoAccounting>
                        <label htmlFor="where-accounting">Куда было потрачено</label>
                        <InputAccounting id="where-accounting" type={"text"}/>
                    </InputInfoAccounting>
                    <InputInfoAccounting>
                        <label htmlFor="how-much-accounting">Сколько было потрачено</label>
                        <InputAccounting id="how-much-accounting" type={"number"} width={"180px"}/>
                    </InputInfoAccounting>
                    <ButtonAccounting/>
                </Managing>
                <Sum>{ `Итого: ${sum} р.` }</Sum>
                <ViewAccounting id="view-accounting">
                    {
                        accountings.map((val, index) => 
                            {
                                return <Accounting 
                                            key={val._id}
                                            _id={val._id}
                                            where={val.where}
                                            date={val.date}
                                            howMuch={val.howMuch}
                                            index={index}
                                        />
                            }
                        )
                    }
                </ViewAccounting>
            </Content>
        </MainWrapper>
    )
}

export default ToDoList;