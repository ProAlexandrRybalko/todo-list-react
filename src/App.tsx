import { useEffect, useState } from 'react';
import { Route, Routes, useNavigate } from 'react-router-dom';
import Home from "./components/Home/Home";
import Login from './components/Login/Login';
import Register from './components/Register/Register';
import ToDoList from './components/ToDoList/ToDoList';

const App = () => {

  const tokenFromSession = (JSON.parse(sessionStorage.getItem('token') as string));
  const [token, setToken] = useState<string>(tokenFromSession);
  const signIn = useNavigate();

  useEffect(() => {
    if(!token) {
      signIn('/login');
    }
  }, []);

    return (
      <>
        {
          token ?
          <Routes>
            <Route path="/" element={<Home/>}/>
            <Route path="/todolist" element={<ToDoList/>}/> 
          </Routes>

          :   null
        }
          <Routes>
            <Route path="/login" element={<Login setToken={setToken}/>}/>
            <Route path="/register" element={<Register setToken={setToken}/>}/>
          </Routes>
      </>

    );
}

export default App;
