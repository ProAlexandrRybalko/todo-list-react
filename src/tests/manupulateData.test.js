import { fireEvent, render, screen } from "@testing-library/react"
import axios from "axios";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import App from "../App";
import { getAllAccountings } from "../assets/api/FetchAllAccountings";
import ToDoList from "../components/ToDoList/ToDoList";
import { store } from "../store";

jest.mock('axios');

describe('getData', () => {
    let response;

    beforeEach(() => {
        response = {
            data: {
                accountings: [
                    {
                        "_id": 0,
                        "where": '',
                        'date': '',
                        'howMuch': ''
                    },
                ],
                sum: 1000,
            }
        }
    })

    test("AddButtonEvent", async () => {
        axios.get.mockReturnValue(response);
        render(
            <MemoryRouter>
                <Provider store={ store }>
                    <ToDoList/>
                </Provider>
            </MemoryRouter>
        );
        const accountings = await screen.findAllByTestId('accounting');
        const todoList = screen.getByTestId('todoList');

        expect(todoList).toMatchSnapshot('toDoList');
        expect(accountings.length).toEqual(1);
        expect(axios.get).toBeCalledTimes(1);
    });
})

