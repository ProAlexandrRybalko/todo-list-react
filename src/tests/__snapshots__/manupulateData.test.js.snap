// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData AddButtonEvent: toDoList 1`] = `
<div
  class="sc-dkPtRN MdcJK"
  data-testid="todoList"
>
  <div
    class="sc-hKwDye cAMyqg"
  >
    <p>
      Учёт моих расходов
    </p>
  </div>
  <div
    class="sc-eCImPb gYpgYl"
  >
    <div
      class="sc-jRQBWg xsnkM"
    >
      <div
        class="sc-iCfMLu gLzwlp"
      >
        <label
          for="where-accounting"
        >
          Куда было потрачено
        </label>
        <input
          class="sc-gKclnd ejFmQy"
          id="where-accounting"
          type="text"
        />
      </div>
      <div
        class="sc-iCfMLu gLzwlp"
      >
        <label
          for="how-much-accounting"
        >
          Сколько было потрачено
        </label>
        <input
          class="sc-gKclnd coHNiZ"
          id="how-much-accounting"
          type="number"
          width="180px"
        />
      </div>
      <div
        class="sc-jrQzAO ueAhs"
      >
        <button
          class="sc-pVTFL bokSBJ"
          data-testid="addButton"
        >
          Добавить
        </button>
      </div>
    </div>
    <div
      class="sc-furwcr etPkRK"
    >
      Итого: 1000 р.
    </div>
    <div
      class="sc-kDTinF fLNJUH"
      id="view-accounting"
    >
      <div
        class="sc-ksdxgE feVBbU"
        data-testid="accounting"
      >
        <div
          class="sc-egiyK fDobeA"
        >
          <p
            class="sc-hBUSln iRESWR"
          >
            1) 
          </p>
        </div>
        <div
          class="sc-iqseJM gyIrNl"
        >
          <p
            class="sc-hBUSln iRESWR"
          >
            NaN.NaN.NaN
          </p>
        </div>
        <div>
          <p
            class="sc-hBUSln iRESWR"
          />
        </div>
        <div
          class="sc-bqiRlB kJCZuL"
        >
          <img
            alt="cross"
            class="sc-crHmcD eKBpnZ"
            src="cross-removebg-preview.png"
          />
          <img
            alt="pencil"
            class="sc-crHmcD eKBpnZ"
            src="pencil-red-eraser-removebg-preview.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
